graph/src/data/store/mod.rs

```diff
@@ -133,6 +133,15 @@ impl FromStr for ValueType {
     }
 }
 
+impl ValueType {
+    /// Return `true` if `s` is the name of a builtin scalar type
+    pub fn is_scalar(s: &str) -> bool {
+        Self::from_str(s)
+            .map(|vt| vt != ValueType::List)
```

Comment:
I'd let the caller use `from_str` and then have this be `fn is_scalar(&self) -> bool`. And to be more future proof exhaustively match on the `ValueType`.

Reply:
I initially had it as `is_scalar(&self)` but by the time you've dealt with the `Result` from `from_str` at the call site, it's much harder to see what is actually being checked by the caller. If this ever gets used from more than one place, we can change it again.

I did make the test for `ValueType::List` and exhaustive `match` for safety; though the existence of `ValueType::List` is very suspect anyway, I think it's only used in one or two places in the code. But that's for another day.